OpenRA.Game/Activities/Activity.cs

```diff
@@ -243,12 +242,18 @@ public virtual void Queue(Activity activity)
 				NextInQueue = activity;
 		}
 
-		public virtual void QueueChild(Activity activity)
+		public virtual void QueueChild(Activity activity, Actor tickActor=null)
```

Comment:
Hmm... I hadn't considered the requirement to pass in the parent actor.

To keep consistency and simplify reviewing IMO we should split out a prereq PR that adds the missing `Actor self` parameter to `Queue`, `QueueChild`, `PrintActivityTree`. This PR can also tweak `ResupplyAircraft` and remove the now-unused `CompositeActivity` to simplify this one a bit more.

I don't mind doing this PR if you do.

Reply:
If we're going that route it makes sense to include all the changes to `Activity` including those that are now duplicated in #16193 and make it a prerequisite for both PRs.

I'm not sure what `Actor self` is supposed to do in `PrintActivityTree`?